# Practica 3 MAD1

**Alumno: Martín Godoy**

**Universidad Nacional del Sur**

Ejercicios sacados del pdf en GitHub MAD1-2026/Unidad 3/U3_regresion_teoria.pdf

---

## Importamos librería

In [ ]:
# Librería necesaria
import numpy as np


---
## Ejercicio 10.1: Regresión Lineal Simple

Consigna: Se midió la altura (X, en cm) y el peso (Y, en kg) de 15 estudiantes. Los datos se resumen en media_muestral(X) = 170, media_muestral(Y) = 68, SXX = 450, SYY = 280, SXY = 330

### 1)  Calcular las estimaciones de β_0 y β_1


In [14]:
SXX = 450
SYY = 280
SXY = 330
media_muestral_X = 170
media_muestral_Y = 68

β_1 = SXY / SXX
β_0 = media_muestral_Y - β_1 * media_muestral_X

print(f"β_1 (pendiente): {β_1}")
print(f"β_0 (intercepto): {β_0}")    

β_1 (pendiente): 0.7333333333333333
β_0 (intercepto): -56.66666666666666


### 2) Interpretar los coeficientes

En nuestro caso β_1 (pendiente) = 0.73 y β_0 (intercepto) = -56.67. Nuestro β_1 = 0.73, nos dice que ante un cambio de altura de 1 centímetro el peso esperado aumenta en 0.73 kg, entonces hay cierta relación entre el peso de una persona y su estatura, por otro lado, nuestro β_0 = -56,67 nos indica cuanto seria el valor esperado de Y (peso) cuando X = 0, sin embargo, tratándose en este caso de peso y altura no tiene mucho sentido analizarlo, pues el peso en ese caso resultaría negativo para una persona que mide 0 cm, que tampoco tiene mucho sentido.

### 3) Calcular R²

In [15]:
#Calculemos el coeficiente de determinación (R^2)
R_squared = (SXY ** 2) / (SXX * SYY)
print(f"Coeficiente de determinación (R^2): {R_squared}")

Coeficiente de determinación (R^2): 0.8642857142857143


### 4) Predecir el peso de un estudiante de 175cm

In [16]:
#Para predecir el valor nos basta con la ecuación de la recta de regresión lineal simple:
#Y = β_0 + β_1 * X
Y_pred = β_0 + β_1 * (175) # Sustituimos X = 175
print(f"Predicción de Y para X=175: {Y_pred}")

Predicción de Y para X=175: 71.66666666666666


---

---

## Ejercicio 10.2: Diagnostico 

Consigna: Ajustaste una regresión lineal y observaste en el gráfico de residuos vs ajustados un patrón en forma de embudo.

### 1) ¿Qué supuesto se viola?

En este caso, donde se aprecia en el gráfico de residuos vs ajustados una forma de embudo, indicaría que se esta violando el supuesto de homocedastisidad, en este caso, implica que hay heterocedastisidad en los errores.

### 2) ¿Qué consecuencias tiene?

Si se violase la homocedastisidad, ocurriría que, a pesar de los estimadores β_0 y β_1 siguen siendo insesgados, estos serian ineficientes, los errores estándares estimados podrían ser incorrectos y como consecuencia los intervalos de confianza, los test t y los test F pueden resultar poco confiables.

### 3) Proponer dos posibles soluciones.

En este caso hay distintas soluciones, las posibilidades podrían ser: Transformaciones de la variable respuesta (Logaritmo, raíz cuadrada), Regresión polinomial o Errores estándares robustos (de White).

---

---

## Ejercicio 10.3: Regresión múltiple

Consigna: Se ajustó un modelo de regresión múltiple para predecir el precio de autos usados (Y) en función de: 
- X1:  antigüedad (años)
- X2:  kilometraje (miles)
- X3:  potencia del motor (HP)

El modelo ajustado es: Y = 25000 − 1500 * X1 − 0.05 * X2 + 100 * X3

### 1) Interpretar cada coeficiente. 

En este caso tenemos los coeficientes β_0 = 25000, β_1 = -1500, β_2 = -0.05 y β_3 = 100, en nuestro caso, nos muestran como afectan ciertas características de un auto a su precio: 
- El β_0 = 25000 nos diría que para un auto con, X1=0 y X2=0 (nuevo) y X3=0 (0 potencia del motor que no tendría mucho sentido) saldría 25000$, que dentro de todo tendría sentido, salvando quizás el X3=0.
- β_1 = -1500 nos dice que el precio del auto, mientras mayor antigüedad en años tenga este mismo, menor sera.
- β_2 = -0.05 nos esta diciendo que el valor del auto es menor a mayor kilometraje, sin embargo este no es un factor tan importante
- β_3 = 100 nos habla de que el valor de auto sera mayor mientras mas potencia tenga el motor.

### 2) Si un auto tiene 3 años, 50.000 km y 150 HP ¿Cuál sera su precio estimado?.

In [17]:
X1 = 3 #(Antigüedad en años)
X2 = 50000 #(Kilometraje en km)
X3 = 150 #(Potencia en HP)

β_0 = 25000
β_1 = -1500
β_2 = -0.05
β_3 = 100

Y_predicted = β_0 + β_1 * X1 + β_2 * X2 + β_3 * X3

print(f"El precio estimado de Y para X1={X1}, X2={X2}, X3={X3}: {Y_predicted}")


El precio estimado de Y para X1=3, X2=50000, X3=150: 33000.0


### 3) ¿Qué variable tiene el mayor impacto absoluto en el precio?

La variable que mas impacto absoluto tiene sobre el precio es la antigüedad en años, pues su coeficiente β_1 = -1500, es decir, por cada año de antigüedad que tenga el auto, su precio va a disminuir por 1500, siendo la variable que mas afectara al precio final de un vehículo 

---

---

## Ejercicio 10.4: Multicolinealidad 

Consigna: Se ajustó un modelo con las variables: Estatura (X1), peso (X2) y circunferencia de la cintura (X3) para predecir la presión arterial. Los VIF son:

VIF1 = 2.3,     VIF2 = 15.2,    VIF3= 14.8


### 1) ¿Hay multicolinealidad? ¿Entre qué variables?

Cuando hablamos de multicolinealidad los valores de VIF>10 son los que nos hablan de multicolinealidad fuerte, en nuestro caso, tenemos que VIF2 y VIF3 son mayores que 10, lo cual implica que las variables X2 y X3 tienen una multicolinealidad con las otras variables respectivamente.

### 2) ¿Qué harías para solucionarlo?

Una de las cosas que se puede hacer para solucionarlo es eliminar una variable redundante, en este caso, me parecería apropiado quitar la variable X3 (circunferencia de la cintura) pues, esta ultima, puede ir bastante de la mano con el peso, entonces me resulta redundante considerarla. 

---

---

## Ejercicio 10.5: Correlación  

Consigna: Se calculó la correlación de Pearson entre horas de estudio (X) y calificación en un examen (Y) para 50 estudiantes, obteniendo r= 0.72.

### 1) Interpretar el valor de r

Tenemos nuestro r = 0.72, como r > 0, nos habla de una correlación positiva entre las horas de estudio y la calificación en un examen, ademas, como r esta bastante cercano a uno, indica que esta correlación es bastante fuerte.

### 2) Realizar el test de significancia (α = 0.05)

Queremos saber si la correlación observada r es suficientemente grande como para concluir que existe en la población una correlación distinta de 0 con H0: p=0 y H1: p≠0 donde p es la correlación poblacional.

In [18]:
#Cuando se conoce r y el tamaño muestral n, se usa la fórmula de la prueba t para el coeficiente de correlación:
r = 0.72
n = 50

t = r * np.sqrt((n - 2) / (1 - r ** 2))

print(f"Estadístico t para el coeficiente de correlación: {t}")

Estadístico t para el coeficiente de correlación: 7.188029916908994


Luego, los grados de libertad son n-2 = 48 con α = 0,05 t₀₀₂₅ ₄₈ ≈ 2.01 entonces como 7.19 > 2.01 se rechaza H0, por tanto existe evidencia significativa  de una correlación entre las horas de estudio y la calificación.

### 3) Calcular R² y explicar su significado

Sabemos que r = Sxy/(Sxx * Syy)^(1/2), pero R² = Sxy²/(Sxx * Syy) por lo tanto r² = R² entonces:

In [19]:
R_squared = r ** 2
print(f"Coeficiente de determinación (R^2): {R_squared} o {R_squared * 100}% ")

Coeficiente de determinación (R^2): 0.5184 o 51.839999999999996% 


Esto se interpreta como que aproximadamente el 51.84% de la variabilidad de las clasificaciones pueden ser explicadas, mediante una relación lineal, por la cantidad de horas que dedico el alumno al estudio.

### 4)¿Podemos concluir que estudiar más causa mejores notas? Justificar.

No puede concluirse una relación causal unicamente a partir de correlación. El valor r nos dice unicamente que las dos variables, horas de estudio y calificación están relacionadas, pero no nos dice que una cause a la otra, es decir, no implica que estudiar sea la causa de obtener mejores notas, podría haber otras variables en juego influyendo como: capacidad del estudiante, calidad de estudio, conocimientos previos, etc.
Para establecer una causalidad seria necesario un análisis que controle también otras variables que podrían influir.